### ISSUE TITLE
Crash: SIGSEGV: Segmentation fault

### ISSUE THREAD

**Issue by DaAngus** [CleverRaven/Cataclysm-DDA#28287]
**Describe the bug**  
What happens is that the game Crashes, log and save file attached.
Some experiments suggest:
Game crashes when a Zombie triggers a landmine under certain conditions. (See Reproduce)
The Game will not crash if:
- I debug teleport to the minefield and teleport back.
- I debug teleport and kite Zombies into the minefield, triggering several mines.
- I run to the east and just fool around.
- I run to the west out of the building, passing time and walk back into the area.
In the last instance a landmine was then triggered without crash.

**To Reproduce**  
This are the inner boundaries under which I am able to reproduce it:
Load the save game and walk in a 2x2 square, don't go further to the east
A feral runner might come, kite and kill inside in close quarters.
The game crashes between 13:36 and 13:40
If not reload. I had ~70% chance to crash.

In Discord nobody was able to reproduce it.

**Screenshots**  
![screenshot from 2019-02-20 15-04-31](https://user-images.githubusercontent.com/45876504/53097289-07091100-3519-11e9-9406-c3d392a4e565.jpg)

**Versions and configuration(please complete the following information):**
 - OS: Arch linux 
 - Game Version: Compiled from source 9. Feb. and 20. Feb.
    https://github.com/CleverRaven/Cataclysm-DDA/
    make -j4 CLANG=1 CCACHE=1 NATIVE=linux64 RELEASE=1 TILES=1 
 - Graphics version: Tiles
 - Mods loaded: none, unmodified base

**Additional context**  
`CRASH LOG FILE: ./config/crash.log
VERSION: 0.C
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:

	./cataclysm-tiles(_Z21debug_write_backtraceRSo+0x3a) [0x5580c2309d7d]
	./cataclysm-tiles(+0x5601b9) [0x5580c22f61b9]
	./cataclysm-tiles(+0x55fdf8) [0x5580c22f5df8]
	/usr/lib/libc.so.6(+0x37e00) [0x7f91949e7e00]
	./cataclysm-tiles(_ZNSt6vectorI8tripointSaIS0_EE12emplace_backIJRiS4_RKiEEEvDpOT_+0x19) [0x5580c2363aeb]
	./cataclysm-tiles(_ZN4game8shrapnelERK8tripointiifi+0x2c2) [0x5580c23625d0]
	./cataclysm-tiles(_ZN4game9explosionERK8tripointRK14explosion_data+0x38d) [0x5580c2361d31]
	./cataclysm-tiles(_ZN4game9explosionERK8tripointffbif+0x54) [0x5580c236195e]
	./cataclysm-tiles(_ZN8trapfunc8landmineEP8CreatureRK8tripoint+0xb9) [0x5580c29c246f]
	./cataclysm-tiles(_ZNK4trap7triggerERK8tripointP8Creature+0x33) [0x5580c29bb8f9]
	./cataclysm-tiles(_ZN7monster7move_toERK8tripointbf+0xb98) [0x5580c270a786]
	./cataclysm-tiles(_ZN7monster4moveEv+0x1236) [0x5580c27089ac]
	./cataclysm-tiles(_ZN4game7monmoveEv+0x759) [0x5580c23b2759]
	./cataclysm-tiles(_ZN4game7do_turnEv+0x746) [0x5580c23b0f38]
	./cataclysm-tiles(main+0x1227) [0x5580c25a682f]
	/usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7f91949d4223]
	./cataclysm-tiles(_start+0x2e) [0x5580c21965be]

	Attempting to repeat stack trace using debug symbols...
	debug_write_backtrace(std::ostream&)
	??:?
	log_crash(char const*, char const*)
	crash.cpp:?
	signal_handler(int)
	crash.cpp:?
	??
	??:0
	void std::vector<tripoint, std::allocator<tripoint> >::emplace_back<int&, int&, int const&>(int&, int&, int const&)
	??:?
	game::shrapnel(tripoint const&, int, int, float, int)
	??:?
	game::explosion(tripoint const&, explosion_data const&)
	??:?
	game::explosion(tripoint const&, float, float, bool, int, float)
	??:?
	trapfunc::landmine(Creature*, tripoint const&)
	??:?
	trap::trigger(tripoint const&, Creature*) const
	??:?
	monster::move_to(tripoint const&, bool, float)
	??:?
	monster::move()
	??:?
	game::monmove()
	??:?
	game::do_turn()
	??:?
	main
	??:?
	__libc_start_main
	??:?
	_start
	??:?
`
[Santa Monica.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/2884919/Santa.Monica.zip)
This world has been reseted after every Death and contains 1 Character
 
I did the report to the best of my knowledge and it took some time, so go easy on me when I missed something obvious. 
It is nothing but a hiccup for me and easily avoidable.

**Comment by zyywo** [comment]
我也遇到了同样的问题，在手持物品栏内的物品时程序崩溃了
Google: I had the same problem, game broke down while I was wielding an item in the inventory.

**To Reproduce**
w - b - j


**Versions and configuration(please complete the following information):**
- OS: Debian9
- Game Version: 0.C-37518-gf76242ff62
- Graphics version: Tiles
- Mods loaded: 禁用NPC需求，肮脏衣物，简单营养

```
The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: ./config/crash.log
VERSION: 0.C-37518-gf76242ff62
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:

	./cataclysm-tiles(_Z21debug_write_backtraceRSo+0x23) [0x9f29c0]
	./cataclysm-tiles() [0x9dfb1a]
	./cataclysm-tiles() [0x9df720]
	/lib/x86_64-linux-gnu/libc.so.6(+0x33060) [0x7f784311c060]
	./cataclysm-tiles(_ZN4itemC2ERKS_+0x14) [0x82e0ce]
	./cataclysm-tiles(_ZNSt7__cxx114listI4itemSaIS1_EE14_M_create_nodeIJRKS1_EEEPSt10_List_nodeIS1_EDpOT_+0x20) [0x8304a0]
	./cataclysm-tiles(_ZNSt7__cxx114listI4itemSaIS1_EE22_M_initialize_dispatchISt20_List_const_iteratorIS1_EEEvT_S7_St12__false_type+0x1c) [0x830462]
	./cataclysm-tiles(_ZNSt7__cxx114listI4itemSaIS1_EEC2ERKS3_+0x2b) [0x82e41f]
	./cataclysm-tiles(_ZN4itemC2ERKS_+0x2d) [0x82e0e7]
	./cataclysm-tiles(_ZNSt7__cxx114listI4itemSaIS1_EE14_M_create_nodeIJRKS1_EEEPSt10_List_nodeIS1_EDpOT_+0x20) [0x8304a0]
	./cataclysm-tiles(_ZNSt7__cxx114listI4itemSaIS1_EE22_M_initialize_dispatchISt20_List_const_iteratorIS1_EEEvT_S7_St12__false_type+0x1c) [0x830462]
	./cataclysm-tiles(_ZNSt7__cxx114listI4itemSaIS1_EEC2ERKS3_+0x2b) [0x82e41f]
	./cataclysm-tiles(_ZN4itemC2ERKS_+0x2d) [0x82e0e7]
	./cataclysm-tiles(_ZNK13item_location4impl14item_on_person11obtain_costERK9Characterl+0x55) [0xc03fa7]
	./cataclysm-tiles(_ZN4game5wieldER13item_location+0x161) [0xad1daf]
	./cataclysm-tiles(_ZN4game5wieldEv+0x34) [0xad1ef0]
	./cataclysm-tiles(_ZN4game13handle_actionEv+0x1111) [0xb110c5]
	./cataclysm-tiles(_ZN4game7do_turnEv+0x487) [0xa9b4cf]
	./cataclysm-tiles(main+0x11f7) [0xc9ac87]
	/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1) [0x7f78431092e1]

	Attempting to repeat stack trace using debug symbols...
	debug_write_backtrace(std::ostream&)
	??:?
	log_crash(char const*, char const*)
	src/crash.cpp:?
	signal_handler(int)
	src/crash.cpp:?
	??
	??:0
	item::item(item const&)
	??:?
	std::_List_node<item>* std::__cxx11::list<item, std::allocator<item> >::_M_create_node<item const&>(item const&)
	??:?
	void std::__cxx11::list<item, std::allocator<item> >::_M_initialize_dispatch<std::_List_const_iterator<item> >(std::_List_const_iterator<item>, std::_List_const_iterator<item>, std::__false_type)
	??:?
	std::__cxx11::list<item, std::allocator<item> >::list(std::__cxx11::list<item, std::allocator<item> > const&)
	??:?
	item::item(item const&)
	??:?
	std::_List_node<item>* std::__cxx11::list<item, std::allocator<item> >::_M_create_node<item const&>(item const&)
	??:?
	void std::__cxx11::list<item, std::allocator<item> >::_M_initialize_dispatch<std::_List_const_iterator<item> >(std::_List_const_iterator<item>, std::_List_const_iterator<item>, std::__false_type)
	??:?
	std::__cxx11::list<item, std::allocator<item> >::list(std::__cxx11::list<item, std::allocator<item> > const&)
	??:?
	item::item(item const&)
	??:?
	item_location::impl::item_on_person::obtain_cost(Character const&, long) const
	??:?
	game::wield(item_location&)
	??:?
	game::wield()
	??:?
	game::handle_action()
	??:?
	game::do_turn()
	??:?
	main
	??:?
	__libc_start_main
	??:?
```
**Additional context**
[W1.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/2893777/W1.tar.gz)

**Comment by ifreund** [comment]
@zyywo I know they both say "segmentation fault" but your crash is actually caused by a different issue.  Will be fixed as soon as #28318 is merged.

**Comment by zyywo** [comment]
> @zyywo I know they both say "segmentation fault" but your crash is actually caused by a different issue. Will be fixed as soon as #28318 is merged.

@ifreund Thanks

**Comment by ZhilkinSerg** [comment]
I cannot reproduce it on latest master with provided savegame. Please inform us if you still experience crash.
